driverapi/driverapi.go

```diff
@@ -57,6 +57,9 @@ type Driver interface {
 	// Leave method is invoked when a Sandbox detaches from an endpoint.
 	Leave(nid, eid string) error
 
+	// Restore reconstruct driver struct
+	Restore(nid, eid string, sboxKey string, ifInfo InterfaceInfo, options map[string]interface{}) error
```

Comment:
It should be possible to not add this new method. And reuse CreateEndpoint(..., restore bool) and Join(..., restore bool)